Rules/CommonScripts/ChatCommands.as

```diff
@@ -34,268 +41,339 @@ bool onServerProcessChat(CRules@ this, const string& in text_in, string& out tex
 	//
 	// player.server_setCoins(player.getCoins() + 100);
 	// Adds 100 coins to the player's coins
+	//
+	// 
 	//-----------------END-----------------//
-
+	if(text_in.substr(0, 1) != "!"){return true;}
 	// cannot do commands while dead
-
-	if (player is null)
-		return true;
+	if (player is null){return true;}
 
 	CBlob@ blob = player.getBlob(); // now, when the code references "blob," it means the player who called the command
 
-	if (blob is null)
-	{
-		return true;
-	}
+	if (blob is null){return true;}
 
 	Vec2f pos = blob.getPosition(); // grab player position (x, y)
 	int team = blob.getTeamNum(); // grab player team number (for i.e. making all flags you spawn be your team's flags)
-
-	// MODDERS --- WRITE ALL COMMANDS BELOW!!
+	uint hash = text_in.getHash();// hash for quick grabs
+	bool isMod = player.isMod();//isMod for quick grabs
 
 	// commands that don't rely on sv_test being on (sv_test = 1)
 
-	if (text_in == "!bot" && player.isMod()) // TODO: whoaaa check seclevs
-	{
-		CPlayer@ bot = AddBot("Henry"); //when there are multiple "Henry" bots, they'll be differentiated by a number (i.e. Henry2)
-		return true;
-	}
-	else if (text_in == "!debug" && player.isMod())
-	{
-		// print all blobs
-		CBlob@[] all;
-		getBlobs(@all);
 
-		for (u32 i = 0; i < all.length; i++)
+	if(isMod)
+	{
+		switch(hash)
 		{
-			CBlob@ blob = all[i];
-			print("[" + blob.getName() + " " + blob.getNetworkID() + "] ");
+			case -149123247://!bot
```

Comment:
It doesn't seem very necessary to me to replace the if-else chain by a switch statement when it implies looking for hashes individually. What would be most ideal is a dictionary to functions (using [funcdefs](https://www.angelcode.com/angelscript/sdk/docs/manual/doc_global_funcdef.html)).
This could potentially help with modularity as well, so you could add commands from other scripts. I don't know how feasible this is in practice though, as behavior across different modules is not very obvious here.